src/pip/_internal/cli/base_command.py

```diff
@@ -91,6 +90,63 @@ def handle_pip_version_check(self, options: Values) -> None:
     def run(self, options: Values, args: List[str]) -> int:
         raise NotImplementedError
 
+    def _run_wrapper(self, level_number: int, options: Values, args: List[str]) -> int:
+        def _inner_run() -> int:
+            try:
+                return self.run(options, args)
+            finally:
+                self.handle_pip_version_check(options)
+
+        if options.debug_mode:
+            rich_traceback.install(show_locals=True)
+            return _inner_run()
```

Comment:
The return value of the command `run()` method is no longer enforced to be an integer, but this seems marginal.

Reply:
Hmm, ignore me. The indirection in the code got me and this inconsistency is present with the old logic as well.

Reply:
Yea, the assert is mostly there for sanity-checking behaviours when working with the CLI but not much else. We could drop it if we have strict mode enabled everywhere relevant.